code/weapon/beam.cpp

```diff
@@ -1136,8 +1139,16 @@ void beam_move_all_post()
 		}
 
 		// add tube light for the beam
-		if(moveup->objp != NULL)
-			beam_add_light(moveup, OBJ_INDEX(moveup->objp), 1, NULL);
+		if (moveup->objp != NULL) {
+			//assume we are in the waiting phases of a type D beam
+			type_d_wait = 0;
+			//test that assumption
+			if (moveup->type == BEAM_TYPE_D)
+				beam_type_d_get_status(moveup, &type_d_index, &type_d_wait);
+			//if it remains true, create a tube light.
+			if (type_d_wait == 0)
```

Comment:
This code is a bit hard to understand since it mixes terminology (e.g. type_d is used even for branches where the beam is not a Type D.

At least for me, it would be easier to understand if the two branches (Type D and non-D) were handled completely separately.

Reply:
would it be clearing if it stayed in the current logic but the variable was renamed "between_type_d_pulses" or the like? To me it would then make plenty of sense that you aren't between pulses in, but if that's still confusing to others it should be an easy structural change

Reply:
My point here is that the code seems to handle different kinds of beam types but you put in a Type D specific check in the middle. For me while reading the code, it is much clearer if you would write something like `if (<type_d> ) <check if between pulses and add light> else <add light>`.

That would make the code flow apparent immediately without having to go back to `type_d_wait` to see that its default value is `0`.